docs/development/pigs/pig-030.rst

```diff
@@ -0,0 +1,447 @@
+            self.parameters = parameters
+            self.effect = effect
+            self.axis = axis
+
+        def __call__(self, irf):
+            if self.axis is None:
+                # Direct modification of data
+                return self.effect(irf, self.parameters)
+
+            # Axis-based coordinate transformation
+            coords = self.effect(irf.geom, self.parameters, self.axis)
+            coords_scaled = MapCoord.create(coords, frame=irf.geom.frame)
+            new_data = irf.interp_by_coord(coords_scaled)
+            return irf.copy(data=new_data)
+
+Effects
+-------
+
+``Effects`` define the mathematical transformation applied to the ``IRF`` *data* or *coordiantes*/*axes*.
+Moreover, ``Effects`` can depend on one or multiple axes, which leads to parameterized transformations, e.g. ``f(enery, offset, ...)``.
+We distinguish between ``ScalingEffects`` and ``ShiftingEffects``.
+
+.. code-block:: python
+
+   class Effect(abc.ABC):
+        def __init__(self, depends_on_axes=None) -> None:
+            self.depends_on_axes = depends_on_axes
+
+        @abc.abstractmethod
+        def __call__(self, data, parameters, apply_to_axes=None):
+            pass
+
+
+    class ScalingEffect(Effect):
+        def __call__(self, data, parameters, apply_to_axes=None):
+            if apply_to_axes is None:
+                if isinstance(parameters, list):
+                    raise ...
+                return data * parameters.value
+            else:
+                coords = data.get_coord()
+                coord_dict = {}
+                for ax, param in zip(coords.axis_names, parameters):
+                    if ax not in apply_to_axes:
+                        coord_dict[ax] = coords[ax]
+                    else:
+                        coord_dict[ax] = coords[ax] * param.value
+                return coord_dict
+
+
+    class ShiftingEffect(Effect):
+        def __call__(self, data, parameters, apply_to_axes=None):
+            if apply_to_axes is None:
+                if isinstance(parameters, list):
+                    raise ...
+                return data + parameters.value
+            else:
+                coords = data.get_coord()
+                coord_dict = {}
+                for ax, param in zip(coords.axis_names, parameters):
+                    if ax not in apply_to_axes:
+                        coord_dict[ax] = coords[ax]
+                    else:
+                        coord_dict[ax] = coords[ax] + param.value
+                return coord_dict
+
+An axis-dependent ``Effect`` can be implemented as
+
+.. code-block:: python
+
+   class LinearScalingEffect(Effect):
+        def __call__(self, data, parameters, apply_to_axes=None):
+            coords = data.get_coord()
+            scaling = 1
+            for i, ax in enumerate(self.depends_on_axes):
+                alpha = parameters[2 * i].value
+                beta = parameters[2 * i + 1].value
```

Comment:
In your example, this represents both shift and scaling, no? It is already the composition of the two effects.